### PR TITLE
ARROW-5077: [Rust] Change Cargo.toml to use release versions

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: dev/release/00-prepare.sh [hunk 1]
```diff
@@ -121,6 +121,27 @@ update_versions() {
...
+  if [ ${type} = "snapshot" ]; then
+    sed -i.bak -E -e \
+      "s/^arrow = { version = \".+\" }/arrow = { path = \"..\/arrow\"\ }/g" \
+      datafusion/Cargo.toml
+    sed -i.bak -E -e \
+      "s/^parquet = { version = \".+\" }/parquet = { path = \"..\/parquet\"\ }/g" \
+      datafusion/Cargo.toml
+    sed -i.bak -E -e \
+      "s/^arrow = { version = \".+\" }/arrow = { path = \"..\/arrow\"\ }/g" \
+      parquet/Cargo.toml
```
Comment (kou) [inline dev/release/00-prepare.sh]: Can we use one `sed`?

```shell
sed -i.bak -E \
  -e "s/^arrow = \".+\"/arrow = { path = \"..\/arrow\" }/g" \
  -e "s/^parquet = \".+\"/parquet = { path = \"..\/parquet\" }/g" \
  */Cargo.toml
```

File: dev/release/00-prepare.sh [hunk 2]
```diff
@@ -121,6 +121,27 @@ update_versions() {
...
+  else
+    sed -i.bak -E -e \
+      "s/^arrow = { path = \"..\/arrow\"\ }/arrow = { version = \"${version}\" }/g" \
+      datafusion/Cargo.toml
+    sed -i.bak -E -e \
+      "s/^parquet = { path = \"..\/parquet\"\ }/parquet = { version = \"${version}\" }/g" \
+      datafusion/Cargo.toml
+    sed -i.bak -E -e \
+      "s/^arrow = { path = \"..\/arrow\"\ }/arrow = { version = \"${version}\" }/g" \
+      parquet/Cargo.toml
```
Comment (kou) [inline dev/release/00-prepare.sh]: Can we use one `sed`?

```shell
sed -i.bak -E \
  -e "s/^arrow = \{ path = \".+\" \}/arrow = \"${version}\"/g" \
  -e "s/^parquet = \{ path = \".+\" \}/parquet = \"${version}\"/g" \
  */Cargo.toml
```

File: dev/release/00-prepare-test.rb [hunk 3]
```diff
@@ -184,6 +184,10 @@ def test_update_version_pre_tag
                      hunks: [
                        ["-version = \"#{@snapshot_version}\"",
                         "+version = \"#{@release_version}\""],
+                       ["-arrow = { path = \"../arrow\" }",
+                        "-parquet = { path = \"../parquet\" }",
+                        "+arrow = { version = \"#{@release_version}\" }",
```
Comment (kou) [inline dev/release/00-prepare-test.rb]: Ah, sorry.
Expected is `arrow = "#{@release_version}"`. We don't need `{ version = ... }`.

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

File: dev/release/00-prepare.sh (hunk 1 above)
Comment (shiro615) [reply dev/release/00-prepare.sh]: Yes. I've fixed it.

File: dev/release/00-prepare-test.rb (hunk 3 above)
Comment (shiro615) [reply dev/release/00-prepare-test.rb]: Thank you for pointing out. I've fixed them.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**